File: deno/lazy.ts
export const createLazy = <T>(
  getFunction: () => T,
): Lazy<T> => {
  let value: T | undefined = undefined;
  return {
    value: () => {
      if (value === undefined) {
        const v = getFunction();
        value = v;
        return v;
      }
      return value;
    },
  };
};

export type Lazy<T> = { readonly value: () => T };

File: dart/commection/lib/lazy.dart
class Lazy<T> {
  Lazy({required this.getFunction}) : _value = null;

  final T Function() getFunction;
  T? _value;

  T get value {
    final value = _value;
    if (value == null) {
      final v = getFunction();
      _value = v;
      return v;
    }
    return value;
  }
}
